enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/throttle/ThrottleFilter.java

```diff
@@ -298,7 +298,9 @@ private Decision checkResourceThrottled(String throttleKey, String tier, Request
             tenantDomain = APIConstants.SUPER_TENANT_DOMAIN_NAME;
         }
 
-        if (!StringUtils.isEmpty(apiTier) && !ThrottleConstants.UNLIMITED_TIER.equals(apiTier)) {
+        // apiConfig instance will have the tier assigned only if openapi definition contains the
+        // extension
+        if (!StringUtils.isEmpty(api.getTier())) {
```

Comment:
Do we need same fix around condition at L138?

Reply:
Since it is unlimited tier, I guess it does not matter which throttle key we set. But, I think it is better to have the fix in L138 as well, since now the apiTier in APIConfig remains empty (with this fix)

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/2556/commits/4f7da1064b42b3726baf0a0b30544284a1159edd

Reply:
One more thing. Do you think we need to change the logic in `getApiTier` method?

Reply:
Nope. It is better to keep it in that way I guess. Because if the API Tier is not present, it means unlimited .